components/compliance-service/api/tests/64_A2V4_migration_spec.rb

```diff
@@ -0,0 +1,273 @@
+      "ICMP ratemask--sysctl-06--",
+      "ICMP ratelimit--sysctl-05--",
+      "ICMP echo ignore broadcasts--sysctl-04--",
+      "ICMP ignore bogus error responses--sysctl-03--" ]
+    assert_suggestions_text_id_version(expected, actual_data)
+
+
+
+    actual_data = GRPC reporting, :list_suggestions, Reporting::SuggestionRequest.new(
+      type: 'control',
+      text: 'sys',
+      filters: [
+        Reporting::ListFilter.new(type: 'profile_id', values: ['b53ca05fbfe17a36363a40f3ad5bd70aa20057eaf15a9a9a8124a84d4ef08015'])
+      ]
+    )
+    expected = [
+      "Magic SysRq--sysctl-30--",
+      "Disable the system`s acceptance of router advertisement--sysctl-25--",
+      "Protection against SYN flood attacks--sysctl-11--",
+      #TODO - the control below should not be here. it is because we get info based on report..but this should only include
+      # controls of the the given profile.
+      "Disable Apache’s follows Symbolic Links for directories in alias.conf--apache-11--"]
+    assert_suggestions_text_id_version(expected, actual_data)
+
+    # suggest profiles with control filters. Should not show profiles that ran on a node without containing the control
+    actual_data = GRPC reporting, :list_suggestions, Reporting::SuggestionRequest.new(
+      type: 'profile',
+      filters: [
+        Reporting::ListFilter.new(type: 'control', values: ['sysctl-06','missing-one'])
+      ]
+    )
+    #TODO - the profile (apache-baseline) below should not be here. it is because we get info based on report..but this
+    # should only include profiles that contain the given control
+    expected = ["DevSec Apache Baseline--41a02784bfea15592ba2748d55927d8d1f9da205816ef18d3bb2ebe4c5ce18a8--2.0.0",
+                "DevSec Linux Security Baseline--b53ca05fbfe17a36363a40f3ad5bd70aa20057eaf15a9a9a8124a84d4ef08015--2.0.1"]
+    assert_suggestions_text_id_version(expected, actual_data)
+
+    # Get a specific report
+    res = GRPC reporting, :read_report, Reporting::Query.new(id: 'bb93e1b2-36d6-439e-ac70-cccccccccc07')
+    assert_equal(Reporting::Report, res.class)
+
+    # in case you need to update the expected_json, print it out with the line below
+    #puts actual_data.to_json
+
+    assert_equal('3.1.3', res['version'])
+    assert_equal('bb93e1b2-36d6-439e-ac70-cccccccccc07', res['id'])
+    assert_equal('9b9f4e51-b049-4b10-9555-10578916e149', res['node_id'])
+    assert_equal('centos(2)-beta-nginx-apache', res['node_name'])
+    assert_equal('DevSec Prod Beta', res['environment'])
+    assert_equal('failed', res['status'])
+    assert_equal(Google::Protobuf::Timestamp.new(seconds: 1520233322, nanos: 0), res['end_time'])
+    assert_equal(Reporting::Platform.new(name: "centos", release: "5.11"), res['platform'])
+    assert_equal(Reporting::Statistics.new(duration: 4.109065055847168), res['statistics'])
+    assert_equal(Google::Protobuf::RepeatedField, res['profiles'].class)
+    if res['profiles'].is_a?(Google::Protobuf::RepeatedField)
```

Comment:
This is probably a vestige from the way these tests were run previously. This conditional is unnecessary because the test on the line above will fail if this isn't true.

Reply:
This line is here to prevent the test from crashing on the lines inside of the 'if'. In golang, I would have a "require.Equals" for the test on line 245.

Reply:
That's exactly what I mean, if the assertion on line 245 fails, line 246 will not be executed. If the assertion on line 245 passes, the condition on line 246 will always be true.

Reply:
Oh, I thought that the test would keep running if an assert fails.
I will remove it then.